mesh/meshdb.go

```diff
@@ -342,6 +386,13 @@ func (m *MeshDB) writeTransactions(txs []*types.Transaction) error {
 		if err := batch.Put(t.Id().Bytes(), bytes); err != nil {
 			return fmt.Errorf("could not write tx %v to database: %v", t.Id().ShortString(), err)
 		}
+		// write extra index for querying txs by account
+		if err := batch.Put(getTransactionOriginKey(l, t), t.Id().Bytes()); err != nil {
+			return fmt.Errorf("could not write tx %v to database: %v", t.Id().ShortString(), err)
+		}
+		if err := batch.Put(getTransactionDestKey(l, t), t.Id().Bytes()); err != nil {
+			return fmt.Errorf("could not write tx %v to database: %v", t.Id().ShortString(), err)
+		}
```

Comment:
I wonder if it wouldn't be simpler to have a single index `account->tx` and add two entries per tx, for the origin and destination.

The way it's used, we always query both and unite the results, isn't it?

Reply:
Thinking about it, there's a bigger problem, this solution doesn't support multiple txs from same src to dst in same layer
As for your comment, I didn't understand your suggested mapping, are you suggesting another field in the value that indicates whether the account is in the source or dst of this tx? or just store a mapping of account -> tx on both src and dst addresses?

Reply:
Ooh, I missed that!

I meant the second option, since we don't need to know if it's the source or destination. We just need to get the list of all TXs that an account is involved in.

Reply:
Wait, that's not true! You're adding the TxId to the key...